File: Go/Arrays & Hashing/Group-Anagrams.go
package main

func main() {

}

func groupAnagrams(strs []string) [][]string {
	anagrams := make(map[[26]int][]string)

	for _, str := range strs {
		counter := [26]int{}

		for _, char := range str {
			counter[char-'a']++
		}

		anagrams[counter] = append(anagrams[counter], str)
	}

	values := [][]string{}

	for _, sameAnagrams := range anagrams {
		values = append(values, sameAnagrams)
	}

	return values
}

File: Python/Arrays & Hashing/Group-Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams = collections.defaultdict(list)

        for str in strs:
            counter = [0] * 26
            for char in str:
                counter[ord(char) - ord("a")] += 1
            anagrams[tuple(counter)].append(str)

        return anagrams.values()
